app/models/miq_queue.rb

```diff
@@ -237,7 +237,12 @@ def self.put_or_update(find_options)
     #   it's unordered nature.
     where_scope = if conds.key?(:args)
                     args = YAML.dump conds.delete(:args)
-                    MiqQueue.where(conds).where(['args = ?', args])
```

Comment:
@kbrock @Fryguy is there a reason we're doing this in the first place?  Why are we using the args as part of the where clause?  I guess what I'm saying, if we need to look at the args of a queue message to uniquely identify it, it feels like we're doing it wrong.